compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -3159,3 +3159,41 @@ class VolatileOnVal()(using Context)
 extends SyntaxMsg(VolatileOnValID):
   protected def msg(using Context): String = "values cannot be volatile"
   protected def explain(using Context): String = ""
+
+class UnusedSymbol(pos: SourcePosition, _msg: String, _actions: List[CodeAction])(using Context)
+extends Message(UnusedSymbolID) {
+  def kind = MessageKind.UnusedSymbol
+
+  override def actions(using Context) = _actions
+  override def msg(using Context) = _msg
+  override def explain(using Context) = ""
+}
+
+object UnusedSymbol {
+    private def removeLocalDefAction(treePos: SourcePosition)(using Context): List[CodeAction] = {
+      val source = treePos.source
+      val sourcePos = treePos.sourcePos
+      val endLine = source.offsetToLine(sourcePos.end - 1)
+      val nextLineOffset = source.lineToOffset(endLine + 1)
+      val startOffset = source.lineToOffset(sourcePos.startLine)
+
+      val pathes = List(
+        ActionPatch(
+          srcPos = sourcePos.withSpan(sourcePos.span.withStart(startOffset).withEnd(nextLineOffset)),
+          replacement = ""
+        ),
+      )
+      List(
+        CodeAction(title = s"Remove unused code",
+          description = None,
+          patches = pathes
+        )
+      )
+    }
+    def imports(treePos: SourcePosition)(using Context): UnusedSymbol = new UnusedSymbol(treePos, i"unused import", List.empty)
+    def localDefs(treePos: SourcePosition)(using Context): UnusedSymbol = new UnusedSymbol(treePos, i"unused local definition", removeLocalDefAction(treePos))
+    def explicitParams(treePos: SourcePosition)(using Context): UnusedSymbol = new UnusedSymbol(treePos, i"unused explicit parameter", List.empty)
+    def implicitParams(treePos: SourcePosition)(using Context): UnusedSymbol = new UnusedSymbol(treePos, i"unused implicit parameter", List.empty)
```

Comment:
**thought:** those two will also be problematic when it comes to create automatic actions because one will also have to update call-sites. 

Also, they are only reported for classes not methods.

Reply:
Right, we'd need to know all call sites, and it is an expensive operation.

What seems like a way of handling this is finding the references on the tooling site, and then compute the diagnostic for the call sites. That would require us to compute additional replacements on tooling site.

Maybe we should consider adding something like we have in lsp: `additionalEdits` which could be populated later on ?

This won't solve the issue with recomputing the edit, but would provide a nice API to handle this case. Another problem that arises it that there is a specific scenario that will fail if we choose this approach:
1. API module in lets say 3.5.0 where we want to remove implicit param in definition.
2. Different module in 3.6.0 which is using api with removed implicit params laeding to an error.
If we let tooling do the refactor, we have to take changes in internal AST structure into consideration, unless we rely on some other parser like Scalameta.

This leads to another idea, which in my opinion can work:
Lets add actionable diagnostic to remove extra param in call site, and then after removing implicit param in definition, we can run extra actionable diagnostics on all call sites. This would solve an issue with different versions.